docs/src/tutorials/child-device-config-mngmnt.md

```diff
@@ -0,0 +1,297 @@
+
+[Step 2 Handle config snapshot request from thin-edge.io](#handle-config-snapshot-request-from-thin-edge.io)
+
+In this step configuration files from a child device will be requested to make them visible in Cumulocity IoT.
+
+[Step 3 Handle config update request from thin-edge.io](#handle-config-update-request-from-thin-edge.io)
+
+In this configuration files from a child device will be updated.
+
+
+### Step 1 Startup / Bootstrap of the child device
+
+In this step a child device with its supported configuration list will be created and will be sent to thin-edge.io during the startup/bootstrap phase of the child device agent.
+
+Therefore some steps have to be taken
+
+1. Create a `c8y-configuration-plugin.toml` this file consists of the supported configuration list i.e. a list of configuration files.
+
+Note: this is a separate file and not the already exsisting `c8y-configuration-plugin.toml` file which is used for configuration management of thin-edge.io.
+
+The child device agent needs to capture the list of configuration files that needs be managed from the cloud in a `c8y-configuration-plugin.toml` file in the same format as specified in the [configuration management documentation](https://thin-edge.github.io/thin-edge.io/html/howto-guides/025_config_management_plugin.html) as follows:
+
+```
+files = [
+    { path = '/path/to/some/config', type = 'config1'},
+]
+```
+* `path` is the full path to the configuration file on the child device file system.
+* `type` is a unique alias for each file entry which will be used to represent that file in Cumulocity IoT.
+
+Example:
+
+```
+files = [
+    { path = '/home/pi/config/config1', type = 'config1'},
+    { path = '/home/pi/config/config2', type = 'config2'},
+]
+```
+Note: these referenced files including the path should be created.
+
+2. Upload this file to thin-edge.io via HTTP: The child device agent needs to upload this file to thin-edge.io with an HTTP PUT request to the URL:
+
+`
+http://{tedge-ip}:8000/tedge/file-transfer/{child-id}/c8y-configuration-plugin
+`
+
+* `{tedge-ip}` is the IP of the thin-edge.io device which is configured as `mqtt.external.bind_address` or `mqtt.bind_address` or `127.0.0.1` if neither is configured.
+* `{child-id}` is the child-device-id
+
+
+
+example: `curl -X PUT --data-binary @/home/pi/config/c8y-configuration-plugin.toml http://127.0.0.1:8000/tedge/file-transfer/child1/c8y-configuration-plugin
+`
+
+3. Notify thin-edge.io about the upload via MQTT
+
+
+Once the upload is complete, the agent should notify thin-edge.io about the upload by sending the following MQTT message:
+
+Topic:
+
+`tedge/{child-d}/commands/res/config_snapshot`
+
+Payload:
+
+`{ "type": "c8y-configuration-plugin”, "path": ”/child/local/fs/path”}`
+
+Example:
+
+Note: For using the `mosquito-pub` command (and later on in this tutorial also the `mosquitto_sub` command), you need to install the  mosquitto clients via `sudo apt-get install mosquitto-clients`
```

Comment:
```suggestion
Note: For using the `mosquito-pub` command (and later on in this tutorial also the `mosquitto_sub` command), you need to install the  mosquitto clients via `sudo apt-get install mosquitto-clients`
```

Move this note below the command and also highlight that any MQTT library/client could be used instead of the `mosquitto` ones.

Reply:
changed this note